src/main/java/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -276,6 +295,23 @@ open class ParallelThreadsRunner(
     override fun createTransformer(cv: ClassVisitor): ClassVisitor {
         return CancellabilitySupportClassTransformer(cv)
     }
+
+    override fun getStateRepresentation(): String? {
+        stateRepresentation?.let { return getMethod(testInstance, it).invoke(testInstance) as String }
+        return null
+    }
+
+    override fun transformTestClass() {
+        super.transformTestClass()
+        initializeTestThreadExecutions()
```

Comment:
Because `TestThreadExecutions` depend on the class loader.
This method is invoked either after class loader initialization or after class loader change.
Better names for the method may be `onClassLoaderChange` or `retransformWithLoader`.
However, it seems to me that test thread executions are strongly connected to the test class, so it is ok to reinitialize them in method named `transformTestClass`